app/qml/form/editors/MMFormTextMultilineEditor.qml

```diff
@@ -61,44 +53,79 @@ MMBaseInput {
-    var realHeight = textArea.y + textArea.contentHeight + 2 * textArea.verticalPadding
-    return realHeight < minHeight ? minHeight : realHeight
-  }
-
-  content: TextArea {
+  inputContent: TextArea {
     id: textArea
 
-    property real verticalPadding: 11 * __dp
-
-    y: textArea.verticalPadding
-    height: contentHeight + textArea.verticalPadding
     width: parent.width
+    height: Math.max( implicitHeight, internal.minHeight )
 
-    text: root.text
+    text: _fieldValue === undefined || _fieldValueIsNull ? '' : _fieldValue
     textFormat: root._fieldConfig['UseHtml'] ? TextEdit.RichText : TextEdit.PlainText
 
-    hoverEnabled: true
-    placeholderTextColor: __style.darkGreyColor
-    color: __style.nightColor
+    topPadding: __style.margin12
+    bottomPadding: __style.margin12
+    leftPadding: __style.margin20
+    rightPadding: __style.margin20
 
-    font: __style.p5
-    wrapMode: Text.WordWrap
+    wrapMode: TextEdit.Wrap
 
-    onLinkActivated: function( link ) {
-      Qt.openUrlExternally( link )
+    font: __style.p5
+    color: {
+      if ( root.editState === "readOnly" ) return __style.nightColor
+      if ( root.editState === "enabled" ) return __style.nightColor
+      if ( root.editState === "disabled" ) return __style.mediumGreyColor
+      return __style.nightColor
     }
+    placeholderTextColor: __style.darkGreyColor
+
+    inputMethodHints: Qt.ImhNoPredictiveText
 
-    onTextChanged: root.editorValueChanged( text, text === "" )
+    readOnly: root.editState !== "enabled"
 
-    // Avoid Android's uncommited text
-    // Could in theory be fixed with `inputMethodComposing` TextInput property instead
-    onPreeditTextChanged: if ( __androidUtils.isAndroid ) Qt.inputMethod.commit()
+    background: Rectangle {
+
+      color: {
+        if ( root.editState !== "enabled" ) return __style.polarColor
+        if ( root.validationState === "error" ) return __style.negativeUltraLightColor
+        if ( root.validationState === "warning" ) return __style.negativeUltraLightColor
+
+        return __style.polarColor
+      }
+
+      border.width: {
+        if ( root.validationState === "error" ) return __style.width2
+        if ( root.validationState === "warning" ) return __style.width2
+        if ( textArea.activeFocus ) return __style.width2
+        if ( textArea.hovered ) return __style.width1
+        return 0
+      }
+
+      border.color: {
+        if ( root.editState !== "enabled" ) return __style.polarColor
+        if ( root.validationState === "error" ) return __style.negativeColor
+        if ( root.validationState === "warning" ) return __style.warningColor
+        if ( textArea.activeFocus ) return __style.forestColor
+        if ( textArea.hovered ) return __style.forestColor
+
+        return __style.polarColor
+      }
+
+      radius: __style.radius12
+    }
+
+    onLinkActivated: ( link ) => Qt.openUrlExternally( link )
```

Comment:
Don't we need a special handler for the newly added project:// links?

Reply:
Not yet, that is in a feature branch for now